lib/random/formatter.rb

```diff
@@ -174,6 +174,128 @@ def uuid
     "%08x-%04x-%04x-%04x-%04x%08x" % ary
   end
 
+  alias uuid_v4 uuid
+
+  # The system clock used by #uuid_v7.  This will be Process::CLOCK_TAI on
+  # systems where it is defined.  Otherwise, this is Process::CLOCK_REALTIME.
+  UUID7_CLOCK = begin
+    Process.clock_gettime(Process::CLOCK_TAI, :float_millisecond)
+    Process::CLOCK_TAI
+  rescue
+    Process::CLOCK_REALTIME
+  end
+
+  # Generate a random v7 UUID (Universally Unique IDentifier).
+  #
+  #   require 'random/formatter'
+  #
+  #   Random.uuid_v7 # => "0188d4c3-16fe-7f96-85c7-242a7aa58f1e"
+  #   Random.uuid_v7 # => "0188d4c3-16fe-744f-86af-38fa04c62bb5"
+  #   Random.uuid_v7 # => "0188d4c3-16fe-764f-b049-c204ce0afa23"
+  #   Random.uuid_v7 # => "0188d4c3-16fe-7085-b14f-ef6415dc6f31"
+  #   #                    |<--sorted-->| |<----- random ---->|
+  #
+  #   # or
+  #   prng = Random.new
+  #   prng.uuid_v7 # => "0188ca51-5e72-7950-a11d-def7ff977c98"
+  #
+  # The version 7 UUID starts with the least significant 48 bits of a 64 bit
+  # Unix timestamp (milliseconds since the epoch) and fills the remaining bits
+  # with random data, excluding the version and variant bits.
+  #
+  # This allows version 7 UUIDs to be sorted by creation time.  Time ordered
+  # UUIDs can be used for better database index locality of newly inserted
+  # records, which may have a significant performance benefit compared to random
+  # data inserts.
+  #
+  # The result contains 74 random bits (9.25 random bytes).
+  #
+  # See draft-ietf-uuidrev-rfc4122bis[https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/]
+  # for details of UUIDv7.
+  #
```

Comment:
@nevans It would be good to note that fixing the random number seed (e.g., by `Kernel#srand`) does not make this method reproducible.

Reply:
Good point.  I'll add a note for that.  If I remember correctly, I had originally written a version that allowed a keyword argument for the timestamp.  But when I changed the code to allow `extra_timestamp_bits`, it didn't seem like it was worth the complexity.

Reply:
What do you think about the following:
```suggestion
  #
  # Note that this method cannot be made reproducable with Kernel#srand, which
  # can only affect the random bits.  But the sorted bits will still be based on
  # Process.clock_gettime.
  #
```

Reply:
Looks good, thanks!

Reply:
@nevans Ah, there are some other ways to fix the random seed than `Kernel#srand`, such as: `Random.new(0).uuid_v7`.

How about this? This PR is already merged, so I will create another PR if you are OK.

> Note that this method cannot be made reproducable because its output includes not only random bits but also timestamp.